stage0/src/main.rs

```diff
@@ -112,6 +112,62 @@ pub unsafe fn jump_to_kernel(entry_point: VirtAddr) -> ! {
     );
 }
 
+/// Trait for abstracting the call to `PVALIDATE` over different page sizes and ranges.
+trait Validator {
+    fn pvalidate(self, validated: Validation) -> Result<(), InstructionError>;
+}
+
+impl Validator for PhysFrame<Size2MiB> {
+    fn pvalidate(self, validated: Validation) -> Result<(), InstructionError> {
+        match pvalidate(
+            self.start_address().as_u64() as usize,
+            SevPageSize::Page2MiB,
+            validated,
+        ) {
+            Err(InstructionError::FailSizeMismatch) => PhysFrame::<Size4KiB>::range(
+                PhysFrame::from_start_address(self.start_address()).unwrap(),
```

Comment:
Can these unwraps cause a panic?

Reply:
I've added a note, but no: the addresses here are aligned to 2 MiB boundaries, which means they're always aligned to 4 KiB as well.

Reply:
Nice. Thanks.